src/tests/test_hiv_testing.py

```diff
@@ -9,15 +9,29 @@
 def test_hiv_testing_covid():
 
     # build population
-    N = 100000
+    N = 100
     pop = Population(size=N, start_date=date(2010, 1, 1))
     # covid disruption is in place
     pop.hiv_testing.covid_disrup_affected = True
     pop.hiv_testing.testing_disrup_covid = True
 
     # evolve population
     pop.hiv_testing.update_hiv_testing(pop, timedelta(days=30))
+    # check that nobody was tested
+    assert sum(pop.get_variable(col.EVER_TESTED)) == 0
 
+
+def test_hiv_testing_before_start():
+
+    # build population
+    N = 100
+    pop = Population(size=N, start_date=date(2003, 1, 1))
+    pop.data[col.ADC] = True
+    # start date is before testing begins
+    pop.hiv_testing.date_start_testing = 2003.5
```

Comment:
One thing to note about the testing start date is that it's a float and we currently only compare the population date year against it (months and days not taken into account) due to comparison incompatibilities between dates and floats. This is an issue across multiple different modules, not just this one, and should probably be addressed separately from this PR.

Reply:
Why not just use a date instead of a float then?